src/elf/types.rs

```diff
@@ -1,178 +1,41 @@
-    pub e_version: Elf64_Word,
-    pub e_entry: Elf64_Addr, /* Entry point virtual address */
-    pub e_phoff: Elf64_Off,  /* Program header table file offset */
-    pub e_shoff: Elf64_Off,  /* Section header table file offset */
-    pub e_flags: Elf64_Word,
-    pub e_ehsize: Elf64_Half,
-    pub e_phentsize: Elf64_Half,
-    pub e_phnum: Elf64_Half,
-    pub e_shentsize: Elf64_Half,
-    pub e_shnum: Elf64_Half,
-    pub e_shstrndx: Elf64_Half,
-}
-
-#[allow(dead_code)]
-pub const PT_NULL: u32 = 0;
-#[allow(dead_code)]
-pub const PT_LOAD: u32 = 1;
-#[allow(dead_code)]
-pub const PT_DYNAMIC: u32 = 2;
-#[allow(dead_code)]
-pub const PT_INTERP: u32 = 3;
-#[allow(dead_code)]
-pub const PT_NOTE: u32 = 4;
-#[allow(dead_code)]
-pub const PT_SHLIB: u32 = 5;
-#[allow(dead_code)]
-pub const PT_PHDR: u32 = 6;
-#[allow(dead_code)]
-pub const PT_TLS: u32 = 7; /* Thread local storage segment */
-#[allow(dead_code)]
-pub const PT_LOOS: u32 = 0x60000000; /* OS-specific */
-#[allow(dead_code)]
-pub const PT_HIOS: u32 = 0x6fffffff; /* OS-specific */
-#[allow(dead_code)]
-pub const PT_LOPROC: u32 = 0x70000000;
-#[allow(dead_code)]
-pub const PT_HIPROC: u32 = 0x7fffffff;
-#[allow(dead_code)]
-pub const PT_GNU_EH_FRAME: u32 = 0x6474e550;
-#[allow(dead_code)]
-pub const PT_GNU_PROPERTY: u32 = 0x6474e553;
-
-#[allow(dead_code)]
-pub const PT_GNU_STACK: u32 = PT_LOOS + 0x474e551;
-
-#[repr(C)]
-pub struct Elf64_Phdr {
-    pub p_type: Elf64_Word,
-    pub p_flags: Elf64_Word,
-    pub p_offset: Elf64_Off,   /* Segment file offset */
-    pub p_vaddr: Elf64_Addr,   /* Segment virtual address */
-    pub p_paddr: Elf64_Addr,   /* Segment physical address */
-    pub p_filesz: Elf64_Xword, /* Segment size in file */
-    pub p_memsz: Elf64_Xword,  /* Segment size in memory */
-    pub p_align: Elf64_Xword,  /* Segment alignment, file & memory */
-}
-
-#[allow(dead_code)]
-pub const PF_X: Elf64_Word = 1;
-#[allow(dead_code)]
-pub const PF_W: Elf64_Word = 2;
-#[allow(dead_code)]
-pub const PF_R: Elf64_Word = 4;
-
-#[repr(C)]
-pub struct Elf64_Shdr {
-    pub sh_name: Elf64_Word,       /* Section name, index in string tbl */
-    pub sh_type: Elf64_Word,       /* Type of section */
-    pub sh_flags: Elf64_Xword,     /* Miscellaneous section attributes */
-    pub sh_addr: Elf64_Addr,       /* Section virtual addr at execution */
-    pub sh_offset: Elf64_Off,      /* Section file offset */
-    pub sh_size: Elf64_Xword,      /* Size of section in bytes */
-    pub sh_link: Elf64_Word,       /* Index of another section */
-    pub sh_info: Elf64_Word,       /* Additional section information */
-    pub sh_addralign: Elf64_Xword, /* Section alignment */
-    pub sh_entsize: Elf64_Xword,   /* Entry size if section holds table */
-}
+pub use libc::Elf64_Addr;
+pub use libc::Elf64_Half;
+pub use libc::Elf64_Off;
+pub use libc::Elf64_Phdr;
+pub use libc::Elf64_Shdr;
+pub use libc::Elf64_Sxword;
+pub use libc::Elf64_Sym;
+pub use libc::Elf64_Word;
+pub use libc::Elf64_Xword;
```

Comment:
these types are supposed to be part of libelf, not libc. Is it coincidental that we get those exposed from libc as well?

Reply:
I understand they are not part of libc the system library. But they are part of `libc` the crate: https://docs.rs/libc/0.2.139/src/libc/unix/linux_like/linux/mod.rs.html

That is by design to the degree I can tell.

Reply:
ah, I see, libc is not just libc :) sounds good then